packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -380,20 +408,24 @@ export class OpenID4VCIClient {
     if (response.errorBody) {
       debug(`Credential request error:\r\n${JSON.stringify(response.errorBody)}`);
       throw Error(
-        `Retrieving a credential from ${this._endpointMetadata?.credential_endpoint} for issuer ${this.getIssuer()} failed with status: ${
+        `Retrieving a credential from ${this._state.endpointMetadata?.credential_endpoint} for issuer ${this.getIssuer()} failed with status: ${
           response.origResponse.status
         }`,
       );
     } else if (!response.successBody) {
       debug(`Credential request error. No success body`);
       throw Error(
-        `Retrieving a credential from ${this._endpointMetadata
+        `Retrieving a credential from ${this._state.endpointMetadata
           ?.credential_endpoint} for issuer ${this.getIssuer()} failed as there was no success response body`,
       );
     }
     return response.successBody;
   }
 
+  public async exportState(): Promise<OpenID4VCIClientState> {
+    return this._state
```

Comment:
Probably wise to clone this, otherwise changes to the exported object would impact current client instance

Reply:
it now returns a string